packages/databyss-ui/components/Editor/state/reducer.js

```diff
@@ -75,11 +69,50 @@ export const setRawHtmlForBlock = (state, block, html) => {
   return nextState
 }
 
+export const correctRangeOffsetForBlock = (state, block, offset) => {
+  const _state = cloneDeep(state)
+  switch (block.type) {
```

Comment:
Don't need the switch, can use  
```
const refEntity = entities(_state, block.type)[block.refId]
refEntity.ranges = refEntity.ranges.map(r => ({ ...r, offset: r.offset + offset }))
```

BUT

looks like we need to get that refEntity below in `getRangesForBlock`, so let's make a fn `getBlockRefEntity(state, block)` so then we can just do:
```
getBlockRefEntity(_state, block).ranges = getBlockRefEntity(_state, block).ranges.map(r => ({ ...r, offset: r.offset + offset }))
```